packages/page-contracts/src/useWeight.ts

```diff
@@ -57,11 +57,16 @@ function useWeightImpl (): UseWeight {
       executionTime,
       isEmpty,
       isValid: isEmpty || isValid,
+      isWeightV2: !isFunction(api.registry.createType('Weight').toBn),
       megaGas: megaGas || BN_ZERO,
       percentage,
       setIsEmpty,
       setMegaGas,
-      weight
+      weight,
+      weightV2: api.registry.createType('WeightV2', {
+        proofSize: 3194304,
```

Comment:
Really not sure it makes sense to hardcode a proofSize. This could be -

- different based on the actual call made
- different based on the actual runtime config

Reply:
how to calculate proof size

Reply:
I believe (needs testing), but you should be able to get (on weights v2 chains) the result from `api.consts.system. blockWeights` which should look similar to -

```
{
  baseBlock: {
    refTime: 6,731,894,000
    proofSize: 0
  }
  maxBlock: {
    refTime: 2,000,000,000,000
    proofSize: 5,242,880
  }
  perClass: {
    normal: {
      baseExtrinsic: {
        refTime: 94,889,000
        proofSize: 0
      }
      maxExtrinsic: {
        refTime: 1,479,905,111,000
        proofSize: 3,879,731
      }
      ...
```

perClass.maxExtrinsic.proofSize _may_ do the trick as a max allowed.

Reply:
Done